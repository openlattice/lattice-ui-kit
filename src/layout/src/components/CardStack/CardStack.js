// @flow

import styled from 'styled-components';

const CardStack = styled.div`
  display: flex;
  flex-direction: column;
  position: relative;

  & > div {
    margin: 10px 0;
  }

  & > div:first-child {
    margin-top: 0;
  }

  & > div:last-child {
    margin-bottom: 0;
  }
`;

export default CardStack;
