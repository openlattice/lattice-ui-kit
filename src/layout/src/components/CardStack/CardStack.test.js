import React from 'react';
import toJson from 'enzyme-to-json';
import { mount } from 'enzyme';

import CardStack from './CardStack';

describe('CardStack', () => {

  describe('snapshots', () => {

    test('Without child', () => {
      const wrapper = mount(<CardStack />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('with > 1 child', () => {
      const wrapper = mount((
        <CardStack>
          <div>child 1</div>
          <div>child 2</div>
        </CardStack>
      ));
      expect(toJson(wrapper)).toMatchSnapshot();
    });

  });

  describe('render', () => {

    test('should render div', () => {
      const wrapper = mount(<CardStack />);
      expect(wrapper.find('div')).toHaveLength(1);
    });

  });

});
