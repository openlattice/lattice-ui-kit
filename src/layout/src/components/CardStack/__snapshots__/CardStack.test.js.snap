// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardStack snapshots Without child 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
}

.c0 > div {
  margin: 10px 0;
}

.c0 > div:first-child {
  margin-top: 0;
}

.c0 > div:last-child {
  margin-bottom: 0;
}

<styled.div>
  <div
    className="c0"
  />
</styled.div>
`;

exports[`CardStack snapshots with > 1 child 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
}

.c0 > div {
  margin: 10px 0;
}

.c0 > div:first-child {
  margin-top: 0;
}

.c0 > div:last-child {
  margin-bottom: 0;
}

<styled.div>
  <div
    className="c0"
  >
    <div>
      child 1
    </div>
    <div>
      child 2
    </div>
  </div>
</styled.div>
`;
